All Languages - update Template S3 Terraform Backend - pages/public_c…

Mx7ca opened the pull request:
…loud/compute/use_object_storage_terraform_backend_state

This PR fixes https://github.com/ovh/docs/issues/5430 in the documentation of all existing languages.

Mx7ca committed:
All Languages - update Template S3 Terraform Backend - pages/public_cloud/compute/use_object_storage_terraform_backend_state

BrenKt committed:
Update guide.de-de.md

BrenKt committed:
Update guide.en-asia.md

BrenKt committed:
Update guide.en-au.md

BrenKt committed:
Update guide.en-ca.md

BrenKt committed:
Update guide.en-gb.md

BrenKt committed:
Update guide.en-ie.md

BrenKt committed:
Update guide.en-sg.md

BrenKt committed:
Update guide.en-us.md

BrenKt committed:
Update guide.es-es.md

BrenKt committed:
Update guide.es-us.md

BrenKt committed:
Update guide.fr-ca.md

BrenKt committed:
Update guide.fr-fr.md

BrenKt committed:
Update guide.it-it.md

BrenKt committed:
Update guide.pl-pl.md

BrenKt committed:
Update guide.pt-pt.md

scraly reviewed (approved):
Top, thank you!

scraly commented:
Thanks for the PR, I will update it to add the information for old terraform version

scraly reviewed (changes requested):
add compatibility with old version

scraly commented:
The modification will only work for new version, not old version. 

Please add this change:


Before terraform version 1.6.0:

```yaml
terraform {
    backend "s3" {
      bucket = "terraform-state-hp"
      key    = "terraform.tfstate"
      region = "gra"
 #or sbg or any activated high performance storage region
      endpoint = "s3.gra.perf.cloud.ovh.net"
      skip_credentials_validation = true
      skip_region_validation = true
    }
}
```

After terraform version 1.6.0:

```yaml
terraform {
    backend "s3" {
      bucket = "terraform-state-hp"
      key    = "terraform.tfstate"
      region = "gra"
      # sbg or any activated high performance storage region
      endpoints = {
        s3 = "https://s3.gra.perf.cloud.ovh.net/"
      }
      skip_credentials_validation = true
      skip_region_validation      = true
      skip_requesting_account_id  = true
    }
}
```

scraly commented:
FYI the information about the terraform version is in the hashicorp doc:
https://developer.hashicorp.com/terraform/language/upgrade-guides#s3-backend

Mx7ca committed:
All languages - add version-restriction for S3-Backend

Mx7ca committed:
All languages - remove scheme from s3 backend endpoint in old version

Mx7ca commented:
@scraly thanks for the link to [OVH's documentation](https://developer.hashicorp.com/terraform/language/upgrade-guides#s3-backend). I've updated the version restriction to all files and also updated the "updated" date again.
Is it now okay?

scraly commented:
Thanks, it's good for me :-) 

The only thing to change is the date of the modification :)

OVH-Team-Guides commented on pages/public_cloud/compute/use_object_storage_terraform_backend_state/guide.de-de.md:
```diff
@@ -1,7 +1,7 @@
 title: 'Using OVHcloud Object Storage as Terraform Backend to store your Terraform state (EN)'
 excerpt: 'Find out how to use an OVHcloud Object Storage as a Terraform Backend to store your Terraform state'
-updated: 2023-07-31
+updated: 2023-11-30
```
```suggestion
updated: 2023-12-04
```
@Mx7ca , could you please update this date on all the versions ? Thank you

OVH-Team-Guides replied:
We will do it on our side

OVH-Team-Guides commented:
Thank you for your contribution @Mx7ca
